extensions/amp-mplayer/0.1/amp-mplayer.js

```diff
@@ -0,0 +1,480 @@
+      this.contentType_ != 'semantic'
+        ? userAssert(
+            element.getAttribute('data-content-id'),
+            'The data-content-id must be specified for %s ' +
+              'data-content-type in <amp-mplayer> %s',
+            this.contentType_,
+            element
+          )
+        : '';
+
+    this.initSemanticFields_();
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+
+    // Warn if the player does not have video interface support
+    this.readyTimeout_ /*** NECESSARY?? ***/ /** @type {number} */ = Services.timerFor(
+      window
+    ).delay(function() {
+      user().warn(
+        TAG,
+        'Did not receive ready callback from player, ' +
+          'ensure it has the videojs-amp-support plugin.'
+      );
+    }, 3000);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    //Define which layouts our element support
+
+    //Size-defined layouts: fixed, fixed height, responsive and fill
+    return isLayoutSizeDefined(layout);
+    //return layout == Layout.RESPONSIVE;
+  }
+
+  /**
+   * Init Semantic params
+   * @private
+   */
+  initSemanticFields_() {
+    const {element} = this;
+    this.scannedElement_ = element.getAttribute('data-scanned-element') || '';
+    this.tags_ = element.getAttribute('data-tags') || '';
+    this.minimumDateFactor_ =
+      element.getAttribute('data-minimum-date-factor') || '';
+    //console.log(element.getAttribute('data-scanned-element-type'));
+    this.scannedElementType_ =
+      element.getAttribute('data-scanned-element-type') || '';
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  handleMPlayerMessage_(event) {
+    if (
+      !originMatches(
+        event,
+        this.iframe_,
+        'https://syringe.s3-us-west-2.amazonaws.com'
+      )
+    ) {
+      return;
+    }
+    const eventData = getData(event);
+    if (!isJsonOrObj(eventData)) {
+      return;
+    }
+    const data = objOrParseJson(eventData);
+    //console.log(data); DONT GET HERE
+    if (data === undefined) {
+      return; // We only process valid JSON.
+    }
+
+    const eventType = data['event'];
+    //console.log(eventType);
+    if (eventType == 'ready') {
+      /*** NECESSARY?? ***/
+      Services.timerFor(this.win).cancel(this.readyTimeout_);
+    }
+    redispatch(this.element, eventType, {
+      'playing': VideoEvents.PLAYING,
+      'paused': VideoEvents.PAUSE,
+      'muted': VideoEvents.MUTED,
+      'unmuted': VideoEvents.UNMUTED,
+      'ended': [VideoEvents.ENDED, VideoEvents.PAUSE],
+      'ads-ad-started': VideoEvents.AD_START,
+      'ads-ad-ended': VideoEvents.AD_END,
+    });
+  }
+
+  /**
+   * Build Iframe source
+   * @return {string}
+   * @private
+   */
+  iframeSource_() {
+    const {element} = this;
+    const source =
```

Comment:
I used it later in that function. 
I didn't use it at the beginning because I don't want that irrelevant params would appear in the url. Is that ok?

Reply:
You can do something like:

```js
const params = dict({});

if (foo) {
  params['bar'] = 'baz';
}
```

Note that it's important to use `object.['bracketQuotes']` instead of `object.periodProperty` so it gets minified correctly.

Reply:
I changed it. Now its definitely more readable. Can you please take a look and tell me if its ok now?
Thanks!